File: 62-unique-paths/62-unique-paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int N = m + n - 2;
        int R = Math.min(m - 1, n - 1);
        double result = 1;
        for(int i = 1; i <= R; ++i)
            result = result * (N - R + i) / i;
        return (int)result;
    }
}

File: 62-unique-paths/62-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        result, N, R = 1, m + n - 2, min(m - 1, n - 1)
        for i in range (1, R + 1):
            result = result * (N - R + i) // i
        return result
        